### PR TITLE
Note that `BasicBlock` is just an index

### PR DESCRIPTION
**Pull request by camelid**
r? @RalfJung

### CODE REVIEW

File: compiler/rustc_middle/src/mir/mod.rs [hunk 1]
```diff
@@ -1106,6 +1106,9 @@ rustc_index::newtype_index! {
     /// are edges that go from a multi-successor node to a multi-predecessor node. This pass is
     /// needed because some analyses require that there are no critical edges in the CFG.
     ///
+    /// Note that this type is just a pointer; the actual data that a basic block holds is in
+    /// [`BasicBlockData`].
```
Comment (scottmcm) [inline compiler/rustc_middle/src/mir/mod.rs:1110]: nit: Consider avoiding the word "pointer" here -- maybe "index"?

Great to have the link to `BasicBlockData` in here!

Comment (camelid) [reply compiler/rustc_middle/src/mir/mod.rs:1110]: I thought of that, but it sounded kind of funny to me and I wasn't 100% sure that it was an index and not the key to a hashmap or something. I'll see if I can come up with something better...

Comment (scottmcm) [reply compiler/rustc_middle/src/mir/mod.rs:1110]: It's _definitely_ an index -- note that the type is defined by the `rustc_index::newtype_index!` macro.  (And the `START_BLOCK` constant is index `0`.)

Comment (camelid) [reply compiler/rustc_middle/src/mir/mod.rs:1110]: Thanks! I saw that but I think I didn't connect the dots :)

File: compiler/rustc_middle/src/mir/mod.rs [hunk 2]
```diff
@@ -1106,6 +1106,9 @@ rustc_index::newtype_index! {
     /// are edges that go from a multi-successor node to a multi-predecessor node. This pass is
     /// needed because some analyses require that there are no critical edges in the CFG.
     ///
+    /// Note that this type is just an index into [`Body.basic_blocks`](Body::basic_blocks);
```
Comment (camelid) [inline compiler/rustc_middle/src/mir/mod.rs:1109]: Are there other places where `BasicBlock` is an index?

Comment (RalfJung) [reply compiler/rustc_middle/src/mir/mod.rs:1109]: I don't really understand the question?

Comment (camelid) [reply compiler/rustc_middle/src/mir/mod.rs:1109]: I wasn’t sure if there are other places where it is an index into a vector of blocks and I want to make sure I don’t mislead people by saying that it’s an index into specifically Body when in fact there are other places that hold a vector of blocks that are indexed by BasicBlock. Hope that was clearer!

Comment (RalfJung) [reply compiler/rustc_middle/src/mir/mod.rs:1109]: Ah I see, thanks for clarifying.

I don't think there are other lists this is an index into, but you can audit all `IndexVec<BasicBlock, _>` if you want to be sure.

**Comment by RalfJung** [comment]
Network failure.
@bors retry

**Pull request merged**